apps/web/app/_utils.tsx

```diff
@@ -53,12 +56,13 @@ const _generateMetadataWithoutImage = async (
   getTitle: (t: TFunction<string, undefined>) => string,
   getDescription: (t: TFunction<string, undefined>) => string,
   hideBranding?: boolean,
-  origin?: string
+  origin?: string,
+  pathname?: string
 ) => {
   const h = headers();
-  const pathname = h.get("x-pathname") ?? "";
-  const canonical = buildCanonical({ path: pathname, origin: origin ?? CAL_URL });
-  const locale = h.get("x-locale") ?? "en";
+  const _pathname = h.get("x-pathname") ?? pathname ?? "";
+  const canonical = buildCanonical({ path: _pathname, origin: origin ?? CAL_URL });
+  const locale = h.get("x-locale") ?? (await getLocale(buildLegacyRequest(h, cookies()))) ?? "en";
```

Comment:
routes needed to go through middleware for these 2 special headers: `x-pathname` and `x-locale`.
However, getLocale can replace `x-locale` and hard-coded pathname (passed as arg) can replace `x-pathname`